huaweicloud/services/er/resource_huaweicloud_er_vpc_attachment.go

```diff
@@ -0,0 +1,298 @@
+			},
+		},
+	}
+}
+
+func resourceVpcAttachmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.ErV3Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating ER v3 client: %s", err)
+	}
+
+	opts := vpcattachments.CreateOpts{
+		VpcId:               d.Get("vpc_id").(string),
+		SubnetId:            d.Get("subnet_id").(string),
+		Name:                d.Get("name").(string),
+		Description:         d.Get("description").(string),
+		AutoCreateVpcRoutes: utils.Bool(d.Get("auto_create_vpc_routes").(bool)),
+		Tags:                utils.ExpandResourceTags(d.Get("tags").(map[string]interface{})),
+	}
+	instanceId := d.Get("instance_id").(string)
+	resp, err := vpcattachments.Create(client, instanceId, opts)
+	if err != nil {
+		return diag.Errorf("error creating VPC attachment: %s", err)
+	}
+	d.SetId(resp.ID)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"PENDING"},
+		Target:       []string{"COMPLETED"},
+		Refresh:      vpcAttachmentStatusRefreshFunc(client, instanceId, d.Id(), []string{"available"}),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        5 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return resourceVpcAttachmentRead(ctx, d, meta)
+}
+
+func resourceVpcAttachmentRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		config       = meta.(*config.Config)
+		region       = config.GetRegion(d)
+		instanceId   = d.Get("instance_id").(string)
+		attachmentId = d.Id()
+	)
+
+	client, err := config.ErV3Client(region)
+	if err != nil {
+		return diag.Errorf("error creating ER v3 client: %s", err)
+	}
+
+	resp, err := vpcattachments.Get(client, instanceId, attachmentId)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "ER VPC attachment")
+	}
+
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("vpc_id", resp.VpcId),
+		d.Set("subnet_id", resp.SubnetId),
+		d.Set("name", resp.Name),
+		d.Set("description", resp.Description),
+		d.Set("auto_create_vpc_routes", resp.AutoCreateVpcRoutes),
+		d.Set("tags", utils.TagsToMap(resp.Tags)),
+		d.Set("status", resp.Status),
+		d.Set("created_at", resp.CreatedAt),
+		d.Set("updated_at", resp.UpdatedAt),
+	)
+
+	if mErr.ErrorOrNil() != nil {
+		return diag.Errorf("error saving VPC attachment (%s) fields: %s", d.Id(), mErr)
+	}
+	return nil
+}
+
+func updateVpcAttachmentBasicInfo(ctx context.Context, client *golangsdk.ServiceClient, d *schema.ResourceData) error {
+	var (
+		opts = vpcattachments.UpdateOpts{}
+
+		instanceId   = d.Get("instance_id").(string)
+		attachmentId = d.Id()
+	)
+
+	if d.HasChange("name") {
+		opts.Name = d.Get("name").(string)
+	}
+	if d.HasChange("description") {
+		opts.Description = utils.String(d.Get("description").(string))
```

Comment:
can we set Name and Description directly?

Reply:
done.